### PR TITLE
`Development`: Add documentation about markdown support

### PR DESCRIPTION
**Pull request by stefanwaldhauser**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

Documentation about Artemis' markdown support is missing.

### Description

Add documentation about markdown support.

### Steps for Testing

1. Look at the new documentation about markdown support: https://artemis-platform--5435.org.readthedocs.build/en/5435/user/markdown-support/

#### Code Review
- [] Review 1
- [] Review 2

### CODE REVIEW

File: docs/user/markdown-support.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+Markdown Support
+================
+
+.. contents:: Content of this document
+    :local:
+    :depth: 2
+
+`Markdown <https://daringfireball.net/projects/markdown/>`__ is an easy-to-read, easy-to-write syntax for formatting plain text.
+
+Artemis extends the basic `Markdown <https://daringfireball.net/projects/markdown/>`__ syntax to support Artemis-specific features. This Artemis flavored Markdown is used to format text content across the platform using an integrated markdown editor.
```
Comment (jschmautzer) [inline docs/user/markdown-support.rst:12]:
```suggestion
Artemis extends the basic `Markdown <https://daringfireball.net/projects/markdown/>`__ syntax to support Artemis-specific features. This Artemis flavored Markdown is used to format text content across the platform using an integrated Markdown editor.
```

File: docs/user/markdown-support.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+Supported Syntax
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The integrated markdown editor uses `Showdown <https://github.com/showdownjs/showdown>`__. A quick description of the supported syntax can be found `here <https://github.com/showdownjs/showdown/wiki/Showdown's-Markdown-syntax>`__.
```
Comment (jschmautzer) [inline docs/user/markdown-support.rst:17]:
```suggestion
The integrated Markdown editor uses `Showdown <https://github.com/showdownjs/showdown>`__. A quick description of the supported syntax can be found `here <https://github.com/showdownjs/showdown/wiki/Showdown's-Markdown-syntax>`__. 
```

File: docs/user/markdown-support.rst [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+The following Showdown extensions are activated:
+
+- `Showdown Katex <https://obedm503.github.io/showdown-katex>`__ to render LaTeX math and AsciiMath using KaTeX.
+- `Showdown Highlight <https://github.com/Bloggify/showdown-highlight>`__ for syntax highlighting in code blocks.
+
+
+Integrated Markdown Editor
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The markdown editor contains a formatting toolbar at the top, allowing users to format text without learning Markdown syntax.
```
Comment (jschmautzer) [inline docs/user/markdown-support.rst:28]:
```suggestion
The Markdown editor contains a formatting toolbar at the top, allowing users to format text without learning Markdown syntax. 
```

**Review by jschmautzer — COMMENTED** [review]
Looks good to me :) 
Only some small issues to be consistent in the documentation with the capitalization.

**Review by Hialus — CHANGES_REQUESTED** [review]
As an MVP this is looking good already. I left some comments for improvements that I might appreciate in this PR or a future PR. Feel free to incorporate them however you want. 
And one more general feedback:
You currently link to 4 different sites explaining Markdown (Markdown, Showdown, Showdown Katex, and Showdown Highlight). It might be beneficial to add a full documentation of the supported Markdown in the future, so the user doesn't have to click through the docs of the libraries. I've seen websites do this quite often, and I personally appreciate it (example [GitHub](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax)), as I then only have a single reference to consult if I need something specific.

**Review by babyygemperor — DISMISSED** [review]
Looks good to me (other than the things already pointed)
No grammatical errors were found (apart from the capitalisation issue already mentioned)

**Review by Marvin-Vitcu — APPROVED** [review]
Documentation LGTM.

Tested on TS1. Link is there.

**Review by manuelmanso — DISMISSED** [review]
Good starting point. @Hialus already left some good suggestions for what could be extended.

**Review by nairaarnez — DISMISSED** [review]
The documentation looks good, it is easy to follow and understandable!

**Review by MetinLamby — DISMISSED** [review]
nice documentation. Understandable and concise!

**Review by jbllhff — APPROVED** [review]
Tested on ts2, looks good!

**Review by luisa-we — APPROVED** [review]
Documentation looks good!
Also tested on TS1, link is there.

**Review by krusche — APPROVED** [review]
Good enough for a first documentation. As discussed, we should extend it in a follow-up

**Pull request merged**